util/ranger/ranger.go

```diff
@@ -102,6 +102,9 @@ func convertPoint(sc *stmtctx.StatementContext, point *point, tp *types.FieldTyp
 			// Ignore the types.ErrOverflow when we convert TypeNewDecimal values.
 			// A trimmed valid boundary point value would be returned then. Accordingly, the `excl` of the point
 			// would be adjusted. Impossible ranges would be skipped by the `validInterval` call later.
+		} else if point.value.Kind() == types.KindMysqlTime && tp.Tp == mysql.TypeTimestamp && terror.ErrorEqual(err, types.ErrWrongValue) {
```

Comment:
Is this the only case that return `types.ErrWrongValue` and can be saftly ignored?
If in some other cases `ErrWrongValue` is also returned, and we ignore the error unexpectedly

Reply:
I think so.

From what I can see, the `ConvertTo(sc, tp)` with `Kind() == types.KindMysqlTime` and `tp.Tp == mysql.TypeTimestamp` can only end up in the part of types/datum.go above, that I also changed in this PR, so I cannot see any other case that would end up being ignored here.

I.e. it is only points of MysqlTime converted to Timestamp that would generate an error in `GetMysqlTime().Convert()` that would be matched here.